napari/layers/shapes/_shape_list.py

```diff
@@ -785,6 +789,56 @@ def inside(self, coord):
         else:
             return None
 
+    def _inside_3d(self, ray_position: np.ndarray, ray_direction: np.ndarray):
```

Comment:
Our shape selection operates on the meshes, so I _think_ this should work for all shape types, but I need to test.